webknossos-datastore/app/com/scalableminds/webknossos/datastore/helpers/DataSetDeleter.scala

```diff
@@ -14,6 +15,22 @@ trait DataSetDeleter extends LazyLogging {
                    dataSetName: String,
                    isInConversion: Boolean = false,
                    reason: Option[String] = None)(implicit ec: ExecutionContext): Fox[Unit] = {
+    @tailrec
+    def deleter(sourcePath: Path, targetPath: Path, retryCount: Int = 0): Fox[Unit] =
+      try {
+        val deduplicatedTargetPath =
+          if (retryCount == 0) targetPath else targetPath.resolveSibling(targetPath.getFileName + s"($retryCount)")
+        val path = Files.move(sourcePath, deduplicatedTargetPath)
+        if (path == null) {
+          throw new Exception("Deleting dataset failed")
+        }
+        logger.info(s"Successfully moved dataset from $sourcePath to $targetPath...")
+        Fox.successful(())
+      } catch {
+        case _: java.nio.file.FileAlreadyExistsException => deleter(sourcePath, targetPath, retryCount + 1)
+        case e: Exception                                => Fox.failure(s"Deleting dataset failed: ${e.toString}", Full(e))
+      }
+
     val dataSourcePath =
       if (isInConversion) dataBaseDir.resolve(organizationName).resolve(".forConversion").resolve(dataSetName)
```

Comment:
I feel like the directory names ".forConversion" and ".trash" are already defined as constants somewhere? If so, it would be cool if they could be reused here.

Reply:
They are not. Should I define it somewhere? If so, where?

Reply:
Yes, maybe a DirectoryConstants (not sure about the name) trait can be created that all relevant services/controllers/traits can extend so the constants become available